Liberty startup issue:  NumberFormatException in com.ibm.ws.kernel.service.util.JavaInfo when running on GraalVM 1.0 RC16

When running JAR generated with mvn clean package from https://github.com/MaartenSmeets/jvmperformance/tree/master/microservicetestcode/MicroProfile
 (using net.wasdev.wlp.maven.parent:liberty-maven-app-parent:2.6.4,
net.wasdev.wlp.maven.parent:liberty-maven-plugin:2.6.4,
io.openliberty:openliberty-runtime:19.0.0.4)

on GraalVM 1.0 RC16 using the following Dockerfile

```
FROM oracle/graalvm-ce:1.0.0-rc16
VOLUME /tmp
ARG JAR_FILE
COPY ${JAR_FILE} app.jar
ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-XX:+UnlockExperimentalVMOptions","-Xmx2G","-Xms2G","-jar","/app.jar"]
```

with
```
docker build -t testimage -f Dockerfile --build-arg JAR_FILE=target/mp-rest-service.jar .
docker run -d -p 8080:8080 testimage
```

My application will not start and I get the following error (docker logs testimage).

```
[ERROR   ] CWWKE0701E: FrameworkEvent ERROR org.osgi.framework.BundleException: Exception in com.ibm.ws.config.internal.WSConfigActivator.start() of bundle com.ibm.ws.config.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:835)
	at [internal classes]
Caused by: java.lang.ExceptionInInitializerError
	at com.ibm.ws.config.admin.internal.UpdateQueue.<init>(UpdateQueue.java:152)
	... 1 more
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.ibm.ws.kernel.service.util.JavaInfo.<init>(JavaInfo.java:87)
	... 1 more
```

Open Liberty with the same code and same base Java version (8u202) does not cause issues on AdoptOpenJDK, OpenJ9 and Oracle JDK. I'm unsure if this is an issue in the Java version number parsing in the Open Liberty code or in GraalVM for providing a wrong Java version. I also created this issue here: https://github.com/oracle/graal/issues/1214

--- a/dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/JavaInfo.java
+++ b/dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/JavaInfo.java
@@ -44,12 +44,12 @@ public class JavaInfo {
         MAJOR = Integer.valueOf(versionElements[i++]);
 
         if (i < versionElements.length)
-            MINOR = Integer.valueOf(versionElements[i++]);
+            MINOR = parseIntSafe(versionElements[i++]);
         else
             MINOR = 0;
 
         if (i < versionElements.length)
-            MICRO = Integer.valueOf(versionElements[i]);
+            MICRO = parseIntSafe(versionElements[i]);
         else
             MICRO = 0;
 
@@ -84,7 +84,7 @@ public class JavaInfo {
                 while ((srloc + len < runtimeVersion.length()) && Character.isDigit(runtimeVersion.charAt(srloc + len))) {
                     len++;
                 }
-                sr = Integer.parseInt(runtimeVersion.substring(srloc, srloc + len));
+                sr = parseIntSafe(runtimeVersion.substring(srloc, srloc + len));
             }
         }
         SERVICE_RELEASE = sr;
@@ -99,7 +99,7 @@ public class JavaInfo {
                 while ((fploc + len < runtimeVersion.length()) && Character.isDigit(runtimeVersion.charAt(fploc + len))) {
                     len++;
                 }
-                fp = Integer.parseInt(runtimeVersion.substring(fploc, fploc + len));
+                fp = parseIntSafe(runtimeVersion.substring(fploc, fploc + len));
             }
         }
         FIXPACK = fp;
@@ -151,5 +151,16 @@ public class JavaInfo {
 
     public static int fixPack() {
         return instance().FIXPACK;
+    }
+
+    /**
+     * @return the integer value of the string, or 0 if the string cannot be coerced to a string
+     */
+    private static int parseIntSafe(String str) {
+        try {
+            return Integer.parseInt(str);
+        } catch (NumberFormatException e) {
+            return 0;
+        }
     }
 }

